packages/babel-parser/test/fixtures/experimental/pipeline-operator/proposal-fsharp-arrow-and-array-in-body/input.js

```diff
@@ -0,0 +1 @@
+x => x |> [y => y + 1] |> z => z * 2
```

Comment:
Done. I may add a `withNonFSharpBodyContext` method to wrap these usages, cuz this pattern comes up a few times here.

Conceptually, this actually happens a decent amount in the parser: save values of current state, set them to new values, parse, reset to old values. Dunno if this is worthwhile to do in this PR, but a generic function for this would be useful.

Reply:
I have been working on adding something to abstract this pattern which will then be transpiled to normal assignments to avoid the performance overhead

Reply:
Ok, cool, so I'll leave it as-is for now.